components/LoginBtn/LoginBtn.tsx

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useContext, useState } from 'react';
+import { ERC725, ERC725JSONSchema } from '@erc725/erc725.js';
+import erc725Schema from '@erc725/erc725.js/schemas/LSP3UniversalProfileMetadata.json';
+import identicon from 'ethereum-blockies-base64';
+
+import { WalletAddressContext } from '../../contexts/WalletAddressContext';
+import useWalletConnect from '../../hooks/useWalletConnect';
+import { IPFS_GATEWAY_BASE_URL, L16_CHAIN_ID } from '../../constants';
+import { toast } from 'react-toastify';
+import walletConnectConnector from '../../utils/walletConnectConnector';
+import Link from 'next/link';
+import { LSP3Profile } from '../../interfaces/lsps';
+import useWeb3Provider from '../../hooks/useWeb3Provider';
+import { validateLSP3 } from '../../utils/validateLSP3';
+
+const LoginBtn: React.FC = () => {
+  const { setWalletAddress, walletAddress } = useContext(WalletAddressContext);
+
+  const [lsp3JSON, setLsp3JSON] = useState<LSP3Profile>();
+
+  useWalletConnect();
+  const web3Provider = useWeb3Provider();
+
+  const isValidChainId = () => {
+    const chainId = parseInt(window.ethereum.networkVersion);
+    if (chainId && window.ethereum.networkVersion != L16_CHAIN_ID) {
+      toast('Please switch to L16 network');
+      return false;
+    }
+    return true;
+  };
+
+  const userAvatar = (LSP3ProfileJSON: LSP3Profile): string => {
+    if (LSP3ProfileJSON.LSP3Profile?.profileImage[0]?.url) {
+      const LSP3ProfileImage = LSP3ProfileJSON.LSP3Profile.profileImage[0].url;
+      if (LSP3ProfileImage.includes('ipfs://')) {
+        return `${IPFS_GATEWAY_BASE_URL}${
+          LSP3ProfileImage.split('ipfs://')[1]
+        }`;
+      }
+
+      return LSP3ProfileJSON?.LSP3Profile?.profileImage[0].url;
+    }
+    return identicon(walletAddress);
+  };
+
+  const isConnected = async () => {
+    try {
+      const accounts = await window.ethereum.request({
+        method: 'eth_accounts',
+      });
+      if (accounts.length) {
+        isValidChainId() && setWalletAddress(accounts[0]);
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const fetchUPInfos = async () => {
+    const config = { ipfsGateway: IPFS_GATEWAY_BASE_URL };
+    const erc725 = new ERC725(
+      erc725Schema as ERC725JSONSchema[],
+      walletAddress,
+      web3Provider,
+      config,
+    );
+    const LSP3Profile = await erc725.fetchData(['LSP3Profile']);
+    const formattedLSP3 = validateLSP3(LSP3Profile[0].value);
+    setLsp3JSON(formattedLSP3);
+  };
```

Comment:
i think these utils function should be put outside of the component, as they can be used elsewhere. Not just here. You could import this function here